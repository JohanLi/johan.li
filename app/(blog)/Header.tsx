'use client'

import { usePathname } from 'next/navigation'
import Link from 'next/link'

export default function Header() {
  const pathname = usePathname()
  const isHomePage = pathname === '/'

  if (isHomePage) {
    return (
      <Link
        href="/"
        className="inline-block text-4xl font-extrabold tracking-tight text-[#a675a2] md:text-5xl"
      >
        <h1>Johan Li</h1>
      </Link>
    )
  }

  return (
    <Link
      href="/"
      className="text-2xl font-extrabold tracking-tight text-[#a675a2] md:text-3xl"
    >
      <h2>Johan Li</h2>
    </Link>
  )
}
