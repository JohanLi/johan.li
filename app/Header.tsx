'use client'

import { usePathname } from 'next/navigation'
import Link from '../src/components/Link'

export default function Header() {
  const pathname = usePathname()
  const isHomePage = pathname === '/'

  return (
    <Link href="/">
      {isHomePage && (
        <h1 className="text-4xl font-extrabold tracking-tight md:text-5xl text-[#a675a2]">
          Johan Li
        </h1>
      )}
      {!isHomePage && (
        <h2 className="text-2xl font-extrabold tracking-tight md:text-3xl text-[#a675a2]">
          Johan Li
        </h2>
      )}
    </Link>
  )
}
