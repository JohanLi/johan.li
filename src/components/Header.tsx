import React, { ReactElement } from 'react';
import { useLocation, Link } from 'react-router-dom';

import styles from './header.scss';

const Header = (): ReactElement => {
  const { pathname } = useLocation();

  let title = <>Johan Li</>;

  if (pathname === '/') {
    title = <h1>{title}</h1>;
  }

  return (
    <header>
      <Link to="/" className={styles.logo}>
        {title}
      </Link>
    </header>
  );
};

export default Header;
