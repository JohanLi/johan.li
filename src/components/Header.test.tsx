import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Header from './Header';

test('logo on the homepage contains an h1 element', () => {
  const tree = create(
    <MemoryRouter initialEntries={['/']}>
      <Header />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('logo is not an h1 element outside the homepage', () => {
  const tree = create(
    <MemoryRouter initialEntries={['/some-article']}>
      <Header />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
