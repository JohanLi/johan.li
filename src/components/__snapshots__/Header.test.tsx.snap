// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logo is not an h1 element outside the homepage 1`] = `
<header>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    Johan Li
  </a>
</header>
`;

exports[`logo on the homepage contains an h1 element 1`] = `
<header>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    <h1>
      Johan Li
    </h1>
  </a>
</header>
`;
